rfc/20240610-Tofu-Version-Compatibility.md

```diff
@@ -0,0 +1,106 @@
+# Tofu Version Compatibility
+
+Issue: https://github.com/opentofu/opentofu/issues/1708 <!-- Ideally, this issue will have the "needs-rfc" label added by the Core Team during triage -->
+
+As OpenTofu and Terraform diverge, the concept of "software version" for compatibility becomes murky.
+
+The "software version" is used in a few locations:
+* Feature compatibility with config files in the root project and modules
+* Provider compatibility checks (deprecated but still used)
+* State compatibility ("terraform_version" field)
+
+At the time of writing this RFC, OpenTofu 1.7.x has very similar features to Terraform 1.8.x (though not identical). As the "software version" requirements are updated in the previous examples, the exact requirements become murky. A module may require features from "Terraform 1.7.4", but would be perfectly happy with "OpenTofu 1.6.5" (as an example).
+
+> [!NOTE]
+> As we are unable to inspect the Hashicorp Terraform Repository in depth, all version compatibility must be based on Release Notes / Changelog Entries as well as user reports.
+
+## Proposed Solution
```

Comment:
Maybe I missed where it is said in this RFC, but I couldn't find any place mentioning that the approach is to ignore the version constraint altogether or not. It's also a bit unclear whether this is the approach in this RFC, or if it is to maintain a compatibility table with Terraform versions

Reply:
Hopefully clarified in b3b094db80